.github/workflows/code-coverage.yml

```diff
@@ -59,7 +59,7 @@ jobs:
         source ../scripts/setup-helics-ci-options.sh
         export CTEST_OPTIONS="--output-on-failure"
         echo "Running CI tests with flags ${CI_TEST_FLAGS}"
-        ../scripts/run-ci-tests.sh ${CI_TEST_FLAGS}
+        ../scripts/run-coverage-tests.sh ${CI_TEST_FLAGS}
 
     - name: Gather and upload coverage results
       run: scripts/lcov-helper.sh gather-coverage-info --gcov ${GCOV_TOOL} --codecov
```

Comment:
Lines 5-8, 25, 32, and 58-61 of code-coverage.yml can be removed.

The run tests step can become a single line: `run: ../scripts/run-coverage-tests.sh`